nx/lib/nx/pytorch_backend.ex

```diff
@@ -0,0 +1,177 @@
+
+  @impl true
+  def iota(out, axis \\ nil)
+
+  def iota(%{shape: {}, type: type} = out, nil), do: iota(%{out | shape: {1}}, nil)
+
+  def iota(%{shape: shape, type: type} = out, nil) do
+    t = NIF.arange(0, Nx.size(shape), 1, torch_type(type), shape)
+    from_ref(out, t)
+  end
+
+  def iota(%{shape: {n}, type: type} = out, 0) do
+    t = NIF.arange(0, n, 1, torch_type(type))
+    from_ref(out, t)
+  end
+
+  def iota(%{shape: shape, type: type} = out, axis) do
+    {dims_before, [dim | dims_after]} =
+      shape
+      |> Tuple.to_list()
+      |> Enum.split(axis)
+
+    # Number of repetitions of an index in memory
+    repeat_blocks =
+      dims_after
+      |> Enum.reduce(1, &*/2)
+
+    # Number of cycles of the counting pattern
+    cycles =
+      dims_before
+      |> Enum.reduce(1, &*/2)
+
+    data =
+      for _ <- 1..cycles,
+          i <- 0..(dim - 1),
+          _ <- 1..repeat_blocks,
+          into: "",
+          do: number_to_binary(i, type)
+
+    t = NIF.from_blob(data, shape, torch_type(type))
+    from_ref(out, t)
+  end
+
+  @impl true
+  def eye(%{shape: {n, n}, type: type} = out) do
+    t = NIF.eye(n, torch_type(type))
+    from_ref(out, t)
+  end
+
+  @impl true
+  def random_uniform(%{type: {s, _} = type, shape: shape} = out, min, max) when s in [:u, :s] do
+    tensor_ref = NIF.randint(min, max, shape, torch_type(type))
+    from_ref(out, tensor_ref)
+  end
+
+  def random_uniform(%{type: {:f, _} = type, shape: shape} = out, min, max) do
+    tensor_ref = NIF.rand(min, max, shape, torch_type(type))
+    from_ref(out, tensor_ref)
+  end
+
+  @impl true
+  def random_normal(%{type: type, shape: shape} = out, mu, sigma) do
+    tensor_ref = NIF.normal(mu, sigma, shape, torch_type(type))
+    from_ref(out, tensor_ref)
+  end
+
+  @impl true
+  def from_binary(%{type: type, shape: shape} = out, binary, _opts) do
+    t = NIF.from_blob(binary, shape, torch_type(type))
+    from_ref(out, t)
+  end
+
+  @impl true
+  def to_binary(%{data: %{ref: ref}}, _opts), do: NIF.to_blob(ref)
+
+  ## Shape
+
+  @impl true
+  def reshape(out, %{data: %{ref: ref}} = tensor, shape),
+    do: %{tensor | data: %__MODULE__{ref: NIF.reshape(ref, shape)}}
+
+  @impl true
+  def squeeze(out, tensor, _axes) do
+    t = NIF.squeeze(tensor.data.ref)
+    from_ref(out, t)
+  end
+
+  ## Ops
+
+  @impl true
+  def add(out, left, right) do
+    t = NIF.add(left.data.ref, right.data.ref)
+    from_ref(out, t)
+  end
+
+  @impl true
+  def dot(
+        out,
+        %{type: t1, data: %{ref: left_ref}} = left,
+        _axes1,
+        %{type: t2, data: %{ref: right_ref}} = right,
+        _axes2
+      ) do
+    t = NIF.dot(left_ref, right_ref)
+    from_ref(out, t)
+  end
+
+  @impl true
+  def cholesky(%{type: output_type, shape: {rows, cols}} = out, tensor) do
+    t = NIF.cholesky(tensor.data.ref)
+    from_ref(out, t)
+  end
+
+  @impl true
+  def backend_deallocate(%Nx.Tensor{data: _data}) do
+    :ok
```

Comment:
In this one you are supposed to deallocate the tensor from memory in a way that any future operation with that tensor raises. :)